Add a mail catcher app

sampaccoud opened the pull request:
## Purpose

In non-production environments, we don't want  to send real emails but we want to easily consult the emails sent to anyone.

## Proposal

The solution for this is to use a mail catcher. There are several solutions available and we chose https://github.com/sj26/mailcatcher because it is easy, good enough and distributed as a Docker image.

- [x] Add a separate mailcatcher app not subject to blue/green deployments
- [x] Configure edxapp to send its emails to mailcatcher

jmaupetit commented on apps/mailcatcher/templates/app/dc.yml.j2:
```diff
@@ -0,0 +1,33 @@
+  # When upgrading, we don't need OpenShift to run several pods mounted on the same volume
+  # Instead we need to make sure that the existing pod is switched off before the new one starts.
+  strategy:
+    type: Recreate
```
Since you don't have any volume, this is not required.

jmaupetit commented on group_vars/customer/patient0/development/main.yml:
```diff
@@ -1,6 +1,7 @@
 # Variables specific to the patient0 customer in development env_type
 
 apps:
+  - name: mailcatcher
```
Can you also add it to `env_types` where it is required (_i.e._ `group_vars/env_type/*`)?

sampaccoud replied:
Done

jmaupetit replied:
I only see CI and development, I thought we would use it everywhere but in production?

sampaccoud replied:
@jmaupetit there was no "apps" parameter to add it to so I thought these environments are not yet used anyway.

jmaupetit replied:
Ok fine. In this case, it means that every app is considered as deployable.

sampaccoud committed:
✨(email) add a mailcatcher app to handle emails in each project

On environments other than production, we don't want our
applications to send real emails but we want to be able to
read and act on the emails sent. The best practice to do this
is to install a mail catcher. We will run one instance of
mailcatcher in each project (shared by all apps in this project).

sampaccoud committed:
✨(edxapp) activate mailcatcher for edxapp in patient0

We configure edxapp in patient0 to send its emails to our
mailcatcher app. This is relying on the hawthorn.1-1.0.4
release of our fun-platform image.

sampaccoud committed:
🐛(edxapp) stop overridding SITE_NAME

The version of edxapp hawthorn.1-1.0.4 sets SITE_NAME to
the LMS hosts (resp. CMS) for the lms (resp cms) app. so
we don't need to set it here. It was not set to the correct
value as edxapp expects a domain name, not a display name,
and uses it to build links to the app in the emails it
sends.

sampaccoud committed:
✅(mailcatcher) test that the mailcatcher app starts smoothly

Each app in Arnold is tested by CircleCI via helpers. We deploy
the app and simply test that the route is available and returns
a page with expected content.

jmaupetit commented on apps/mailcatcher/templates/app/svc.yml.j2:
```diff
@@ -0,0 +1,23 @@
+    service: app
+    version: "{{ mailcatcher_image_tag }}"
+    deployment_stamp: "{{ deployment_stamp }}"
+  name: "{{ mailcatcher_app_host }}"
```
I would call it `mailcatcher` and remove this confusing variable (`mailcatcher_app_host` _versus_ `mailcatcher_host`).

sampaccoud replied:
but there is already a "mailcatcher_host" here https://github.com/openfun/arnold/pull/120/files#diff-1d6d10331c2861156e52d71455fd392dR16 so I followed naming from the "redis" app.

jmaupetit replied:
Sorry I did not mean to call your variable `mailcatcher`, but rather hard code the name:

```yaml
name: "mailcatcher"
```

sampaccoud replied:
oh yeah you're right!

jmaupetit commented on apps/mailcatcher/vars/all/main.yml:
```diff
@@ -0,0 +1,11 @@
+# -- mailcatcher
+mailcatcher_image_name: "sj26/mailcatcher"
+mailcatcher_image_tag: "latest"
+mailcatcher_app_host: "mailcatcher"
```
As said, I would remove this var.

jmaupetit reviewed (commented):
This is odd that we cannot see you new CI step (mailcatcher bootstrap)?

sampaccoud commented:
@jmaupetit not weird: I forgot to add it to the list of jobs...

jmaupetit commented:
I should have seen this... well, now it works! :tada:

sampaccoud committed:
🎨(redis) remove variable to configure Redis app host

The Redis app host is internal to our deployments, and making
it configurable is not a feature but a complexity. We can
impose its value to "redis".

jmaupetit reviewed (approved):
Ship it!
